From d26dcd7e7eb2536bb8070fa5891e04c89dbac2a1 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:52 -0400
Subject: [fs] nfsd: clean up helper __release_lock_stateid

Message-id: <1410898849-22126-49-git-send-email-bfields@redhat.com>
Patchwork-id: 95774
O-Subject: [Patch RHEL7 048/165] nfsd: clean up helper __release_lock_stateid
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Use filp_close instead of open coding. filp_close does a bit more than
just release the locks and put the filp. It also calls ->flush and
dnotify_flush, both of which should be done here anyway.

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: e20fcf1e6586ff1620adc345ad2a93d5ee5def59
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 314dc80..4ab567e 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -821,10 +821,8 @@ static void __release_lock_stateid(struct nfs4_ol_stateid *stp)
  unhash_generic_stateid(stp);
  unhash_stid(&stp->st_stid);
  file = find_any_file(stp->st_file);
- if (file) {
-  locks_remove_posix(file, (fl_owner_t)lockowner(stp->st_stateowner));
-  fput(file);
- }
+ if (file)
+  filp_close(file, (fl_owner_t)lockowner(stp->st_stateowner));
  close_generic_stateid(stp);
  free_generic_stateid(stp);
 }
-- 
1.7.1